daml-lf/language/src/main/scala/com/digitalasset/daml/lf/language/Ast.scala

```diff
@@ -712,7 +712,12 @@ object Ast {
       observers: E, // Observers of the contract.
       key: Option[GenTemplateKey[E]],
       implements: Map[TypeConName, GenTemplateImplements[E]],
-  )
+  ) {
+    lazy val inheritedChoices: Map[ChoiceName, TypeConName] =
+      implements.iterator.flatMap { case (iface, impl) =>
+        impl.inheritedChoices.iterator.map ( chName => (chName, iface) )
+      }.toMap
```

Comment:
In this case, the first `.iterator` together with the `.toMap` are not useful because there is not intermediary `Map` that is built. 
In general,  when available, prefer `.view` over `iterator` as the later is pure. 

```suggestion
    lazy val inheritedChoices: Map[ChoiceName, TypeConName] =
      implements.flatMap { case (iface, impl) =>
        impl.inheritedChoices.view.map ( chName => (chName, iface) )
      }
```

Reply:
Thanks!